vscode-web/run.sh

```diff
@@ -2,6 +2,40 @@
 
 BOLD='\033[0;1m'
 EXTENSIONS=("${EXTENSIONS}")
+VSCODE_SERVER="${INSTALL_PREFIX}/bin/code-server"
+
+# Set extension directory
+EXTENSION_ARG=""
+if [ -n "${EXTENSIONS_DIR}" ]; then
+  EXTENSION_ARG="--extensions-dir=${EXTENSIONS_DIR}"
+fi
+
+run_code_server() {
+  echo "👷 Running $VSCODE_SERVER serve-local --port ${PORT} --host 127.0.0.1 --accept-server-license-terms serve-local --without-connection-token --telemetry-level ${TELEMETRY_LEVEL} in the background..."
+  echo "Check logs at ${LOG_PATH}!"
+  "$VSCODE_SERVER" serve-local "$EXTENSION_ARG" --port "${PORT}" --host 127.0.0.1 --accept-server-license-terms serve-local --without-connection-token --telemetry-level "${TELEMETRY_LEVEL}" > "${LOG_PATH}" 2>&1 &
+}
+
+# Check if the settings file exists...
+if [ ! -f ~/.vscode-server/data/Machine/settings.json ]; then
+  echo "⚙️ Creating settings file..."
+  mkdir -p ~/.vscode-server/data/Machine
+  echo "${SETTINGS}" > ~/.vscode-server/data/Machine/settings.json
+fi
+
+# Check if code-server is already installed for offline or cached mode
+if [ -f "$VSCODE_SERVER" ]; then
+  if [ "${OFFLINE}" = true ] || [ "${USE_CACHED}" = true ]; then
```

Comment:
Not a blocker for this PR, but I just realized we might not install extensions (either the ones in the config or the ones in `extensions.json`) if `USE_CACHED` is set, is that how we want it?

Reply:
When cached. When you add a recommended extension, it would be installed or prompt you to install. But it would be installed in the cached persistent store. So stop / start will find it in the cache already.

Reply:
So by design really.

Reply:
I was thinking about a case where someone already has code-server/VS Code Web installed in the image and has the the cache enabled, and because it already exists the script exits early here and never installs the extensions below.

Or, suppose someone is using the cache then adds a new extension to the list.  It will never get installed automatically either.

Reply:
@code-asher  - maybe it can run the installs again but just ignore errors when the same extension already exists.

Reply:
That seems reasonable!

Reply:
(i assume that would be faster than a full install. :) )